projects/community-payback-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/integrations/delius/entity/UnpaidWorkAppointment.kt

```diff
@@ -119,6 +119,25 @@ fun UpwAppointment.toAppointmentResponseCase() = AppointmentResponseCase(
     restrictionMessage = this.person.restrictionMessage,
 )
 
+fun UpwAppointment.toAppointmentResponseCase(
+    currentExclusion: Boolean?,
+    exclusionMessage: String?,
+    currentRestriction: Boolean?,
+    restrictionMessage: String?
+) = AppointmentResponseCase(
+    crn = this.person.crn,
+    name = AppointmentResponseName(
+        forename = this.person.forename,
+        surname = this.person.surname,
+        middleNames = this.person.secondName?.let { names -> listOf(names) } ?: emptyList()
+    ),
+    dateOfBirth = this.person.dateOfBirth,
+    currentExclusion = currentExclusion ?: this.person.currentExclusion,
+    exclusionMessage = exclusionMessage ?: this.person.exclusionMessage,
+    currentRestriction = currentRestriction ?: this.person.currentRestriction,
+    restrictionMessage = restrictionMessage ?: this.person.restrictionMessage,
+)
```

Comment:
Could we merge this with the above method?  We shouldn't be using the fields from the `offender` record any more, only from the `UserAccessService`.  We'll need to add a username parameter into the `getSession` endpoint.

Also - I wonder if we could simplify the `getAppointment` method by just passing the limited access object into here? E.g.
```suggestion
fun UpwAppointment.toAppointmentResponseCase(limitedAccess: CaseAccess) = AppointmentResponseCase(
    crn = this.person.crn,
    name = AppointmentResponseName(
        forename = this.person.forename,
        surname = this.person.surname,
        middleNames = this.person.secondName?.let { names -> listOf(names) } ?: emptyList()
    ),
    dateOfBirth = this.person.dateOfBirth,
    currentExclusion = limitedAccess.userExcluded,
    exclusionMessage = limitedAccess.exclusionMessage,
    currentRestriction = limitedAccess.userRestricted,
    restrictionMessage = limitedAccess.restrictionMessage,
)
```

Reply:
Apologies I missed that we needed it in getSession too so it wasn't in the original story.  Happy to have a quick chat to discuss if you like?